x/ecocredit/server/msg_server.go

```diff
@@ -444,12 +450,7 @@ func (s serverImpl) Cancel(goCtx context.Context, req *ecocredit.MsgCancel) (*ec
 
 // assertClassIssuer makes sure that the issuer is part of issuers of given classID.
 // Returns ErrUnauthorized otherwise.
-func (s serverImpl) assertClassIssuer(goCtx context.Context, classID, issuer string) error {
-	ctx := types.UnwrapSDKContext(goCtx)
-	classInfo, err := s.getClassInfo(ctx, classID)
-	if err != nil {
-		return err
-	}
+func (s serverImpl) assertClassIssuer(classInfo *ecocredit.ClassInfo, issuer string) error {
```

Comment:
You could move this to a method on `ClassInfo` now that you've removed the fetching logic

Reply:
https://github.com/regen-network/regen-ledger/pull/491/commits/5e380b62603390a94c147da06eb9553161c264cb